include/solvers/unsteady_solver.h

```diff
@@ -150,6 +150,38 @@ class UnsteadySolver : public TimeSolver
    */
   virtual bool is_steady() const libmesh_override { return false; }
 
+  /**
+   * Indicate to the UnsteadySolver which variables are first order in time.
+   * Subclasses may use this information in different ways.
+   */
+  void add_first_order_var(unsigned int var)
+  { _first_order_vars.insert(var); }
+
```

Comment:
Second-guessing myself as I read through subsequent commits: you do hit this stuff often enough that "only 2nd order variables" might in fact be worth caching so you don't end up with a bunch of "loop over all variables but skip 1st order" code.